code/menuui/barracks.cpp

```diff
@@ -1508,8 +1508,15 @@ void barracks_do_frame(float frametime)
 				}
 
 				if (Player_sel_mode == PLAYER_SELECT_MODE_SINGLE) {
+					Cur_pilot->flags |= PLAYER_FLAGS_IS_MULTI;
+					Pilot.save_player(Cur_pilot);
 					barracks_init_player_stuff(PLAYER_SELECT_MODE_MULTI);
 				} else {
+					// make sure we don't carry over the multi flag
+					if (Cur_pilot->flags & PLAYER_FLAGS_IS_MULTI) {
```

Comment:
You don't need to check if `PLAYER_FLAGS_IS_MULTI` is already set. `Cur_pilot->flags &= ~PLAYER_FLAGS_IS_MULTI` will have no effect if the flag wasn't present.